internal/runtimehandlerhooks/high_performance_hooks_linux.go

```diff
@@ -631,69 +640,103 @@ func (h *HighPerformanceHooks) setIRQLoadBalancing(ctx context.Context, c *oci.C
 		return fmt.Errorf("find container %s CPUs", c.ID())
 	}
 
-	newIRQBalanceSetting, err := h.updateNewIRQSMPAffinityMask(lspec.Resources.CPU.Cpus, enable)
```

Comment:
Problem:
- If I move the lock here then irqbalance --oneshot and systemctl restart irqbalance are inside the lock; but no race condition.
- If I move the lock into updateNewIRQSMPAffinityMask, the locks guarantee that the SMP affinity file and irqbalance configuration are consistent. But there's a potential race (thread B overtaking thread A after the locked section, thread B running irqbalance --oneshot with new config, thread A running irqbalance --oneshot with the old config
I initially though I could avoid this by writing state to the file, then reading from the file with irqbalance --oneshot. But a) that changed the logic and required the irqbalance configutation file to be present. And b) it wouldn't get rid of the race